OpenSim/Simulation/Test/testFrames.cpp

```diff
@@ -316,88 +332,145 @@ void testPhysicalOffsetFrameOnBodySerialize()
+    for (auto& component :
+         pendulumWFrame->getComponentList<PhysicalOffsetFrame>()) {
         std::cout << "frame[" << ++i << "] is " << component.getName()
-            << " of type " << typeid(component).name() << std::endl;
+        << " of type " << typeid(component).name() << std::endl;
     }
     ASSERT_EQUAL(5, i, 0, __FILE__, __LINE__,
-        "testFilterByFrameType failed to find the 3 PhyscicalOffsetFrame in the model.");
+                 "testFilterByFrameType failed to find the 3 PhyscicalOffsetFrame in the model.");
 }
 
 
+
+
 void testStationOnFrame()
 {
     SimTK::Vec3 tolerance(SimTK::Eps);
-
+    
     cout << "Running testStationOnFrame" << endl;
+    
+    Model* pendulum = new Model("double_pendulum.osim");
+    // Get the frames for the two rods of the pendulum
+    const OpenSim::Body& rod1 = pendulum->getBodySet().get("rod1");
+    const SimTK::Vec3& com1 = rod1.get_mass_center();
+    
+    // Create a station in each frame
+    Station* stationInFrame1 = new Station();
+    stationInFrame1->set_location(com1);
+    stationInFrame1->updConnector<PhysicalFrame>("reference_frame").setConnecteeName("rod1");
+    pendulum->addModelComponent(stationInFrame1);
+    
+    Station* stationInFrame2 = new Station();
+    stationInFrame2->set_location(com1);
+    stationInFrame2->updConnector<PhysicalFrame>("reference_frame").setConnecteeName("rod2");
+    pendulum->addModelComponent(stationInFrame2);
+    
+    Station* stationInGround = new Station();
+    stationInGround->set_location(com1);
+    stationInGround->updConnector<PhysicalFrame>("reference_frame").setConnecteeName("ground");
+    pendulum->addModelComponent(stationInGround);
+    
+    
+    SimTK::Vec3 stationPosition;
+    
+    // Compare the vector between two stations, in two different frames.
+    SimTK::State& s = pendulum->initSystem();
+    for (double ang = 0; ang <= 90.0; ang += 10.){
+        double radAngle = SimTK::convertDegreesToRadians(ang);
+        const Coordinate& coord = pendulum->getCoordinateSet().get("q1");
+        coord.setValue(s, radAngle);
+        
+        // get the position of rod1's station in rod2 frame
+        stationPosition = stationInFrame1->findLocationInFrame(s, pendulum->getBodySet().get("rod2") );
+        // Set the location of the rod2's station.
+        stationInFrame2->set_location(stationPosition);
+        
+        //get the position of rod2's station in ground frame
+        stationPosition = stationInFrame2->findLocationInFrame(s, pendulum->getGround() );
+        // Set the location of the ground's station
+        stationInGround->set_location(stationPosition);
+        
+        //get the position of the ground's station in rod1
+        stationPosition = stationInGround->findLocationInFrame(s, pendulum->getBodySet().get("rod1") );
+        
+        // now compare the original station location in rod1 (com) with the transformed station.
+        SimTK_TEST_EQ(stationPosition, com1);
+    }
```

Comment:
I'm not sure I completely understand this test, but I think you're revisiting it in a separate PR.